developer-cli/Installation/PrerequisitesChecker.cs

```diff
@@ -5,71 +5,36 @@
-        new Prerequisite(PrerequisiteType.CommandLineTool, "docker", "Docker", new Version(27, 1, 1)),
-        new Prerequisite(PrerequisiteType.CommandLineTool, "node", "NodeJS", new Version(22, 3, 0)),
-        new Prerequisite(PrerequisiteType.CommandLineTool, "az", "Azure CLI", new Version(2, 63)),
-        new Prerequisite(PrerequisiteType.CommandLineTool, "gh", "GitHub CLI", new Version(2, 55)),
-        new Prerequisite(PrerequisiteType.DotnetWorkload, "aspire", "Aspire", new Version(8, 2, 0))
-    ];
-
-    public static void Check(params string[] prerequisiteName)
-    {
-        var invalid = false;
-        foreach (var command in prerequisiteName)
-        {
-            var prerequisite = Dependencies.SingleOrDefault(p => p.Name == command);
-            if (prerequisite is null)
-            {
-                AnsiConsole.MarkupLine($"[red]Unknown prerequisite: {command}[/]");
-                invalid = true;
-                continue;
-            }
+    public static readonly Prerequisite Dotnet = new CommandLineToolPrerequisite("dotnet", "dotnet", new Version(8, 0, 300));
+    public static readonly Prerequisite Docker = new CommandLineToolPrerequisite("docker", "Docker", new Version(27, 1, 1));
+    public static readonly Prerequisite Node = new CommandLineToolPrerequisite("node", "NodeJS", new Version(22, 3, 0));
+    public static readonly Prerequisite AzureCli = new CommandLineToolPrerequisite("az", "Azure CLI", new Version(2, 63));
+    public static readonly Prerequisite GithubCli = new CommandLineToolPrerequisite("gh", "GitHub CLI", new Version(2, 55));
+    public static readonly Prerequisite Aspire = new DotnetWorkloadPrerequisite("aspire", "Aspire", new Version(8, 2, 0));
 
-            switch (prerequisite.Type)
-            {
-                case PrerequisiteType.CommandLineTool:
-                    if (!IsCommandLineToolValid(prerequisite.Name, prerequisite.DisplayName, prerequisite.Version))
-                    {
-                        invalid = true;
-                    }
-
-                    break;
-                case PrerequisiteType.DotnetWorkload:
-                    if (!IsDotnetWorkloadValid(prerequisite.Name, prerequisite.DisplayName, prerequisite.Version))
-                    {
-                        invalid = true;
-                    }
-
-                    break;
-            }
-        }
+    protected abstract bool IsValid();
 
-        if (invalid)
+    public static void Ensure(params Prerequisite[] prerequisites)
+    {
+        var isValid = prerequisites.Aggregate(true, (state, prerequisite) => prerequisite.IsValid() && state);
+        if (!isValid)
         {
             Environment.Exit(1);
         }
     }
+}
 
-    private static bool IsCommandLineToolValid(string command, string displayName, Version minVersion)
+file record CommandLineToolPrerequisite(string Command, string DisplayName, Version MinVersion) : Prerequisite
```

Comment:
Could you make these two classes `sealed`. I get a warning in SonarLint.

![CleanShot 2024-09-08 at 16 31 36](https://github.com/user-attachments/assets/7bfe65d9-1ab0-48a6-b3c7-095d6d767279)